hathor/p2p/manager.py

```diff
@@ -176,6 +187,16 @@ def __init__(self, reactor: Reactor, my_peer: PeerId, server_factory: 'HathorSer
         if enable_sync_v2:
             self._sync_factories[SyncVersion.V2] = SyncV1Factory(self)
 
+    def set_manager(self, manager: 'HathorManager') -> None:
+        """Set the manager. This method must be called before start()."""
```

Comment:
Add an `assert self.manager is not None` in `start()`?

Reply:
Or maybe even change the `start()` method to receive the `HathorManager` and set it, instead of having a separate `set_manager()` method

Reply:
Even though I think it is an interesting idea, we usually leave the `start()` and `stop()` method without parameters. This way, we can run them anywhere without requiring access to the dependencies. That's why I used a `set_manager()` method. What do you think?

Reply:
Ok, makes sense!